stroyprombeton/models.py

```diff
@@ -81,13 +82,38 @@ def get_series(self) -> models.QuerySet:
 
 class SeriesQuerySet(models.QuerySet):
 
+    class SQLCount(Subquery):
+        template = "(SELECT count(*) FROM (%(subquery)s) _count)"
+        output_field = IntegerField()
+
     def bind_fields(self):
         """Prefetch or select typical related fields to reduce sql queries count."""
         return self.select_related('page')
 
-    def active(self):
+    def active(self) -> 'SeriesQuerySet':
         return self.filter(page__is_active=True)
 
+    def non_empty(self) -> 'SeriesQuerySet':
+        # @todo #662:30m  Move SQLCount to some utils place.
+        #  `pages.models` module for example.
+        #  Don't forget to run over all "SQLCount" code entries.
+
+        # https://stackoverflow.com/questions/42543978/django-1-11-annotating-a-subquery-aggregate
+        class SQLCount(Subquery):
+            template = "(SELECT count(*) FROM (%(subquery)s) _count)"
+            output_field = IntegerField()
+
+        return (
+            self.active()
```

Comment:
It seems it does the same, but less verbose and simpler
```suggestion
            self.active()
            .filter(options__product__page__is_active=True)
            .distinct()
```